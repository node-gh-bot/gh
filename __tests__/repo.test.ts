/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import { runCmd } from './runCmd'

describe('E2E: Repo Module Test', () => {
    it('List Repos `gh re --list`', done => {
        expect(runCmd('gh re --list')).toMatchSnapshot()
        done()
    })

    it('Create new repo `gh re --new foo --init`', done => {
        expect(runCmd('gh re --new foo --init')).toMatchSnapshot()
        done()
    })

    it('List labels on a repo `gh re --label --list --repo gh`', done => {
        expect(runCmd('gh re --label --list --repo gh')).toMatchSnapshot()
        done()
    })

    it('Fork a repo `gh re --fork prettier --user prettier`', done => {
        expect(runCmd('gh re --fork prettier --user prettier')).toMatchSnapshot()
        done()
    })

    it('Delete repo `gh re --delete foo`', done => {
        expect(runCmd('printf "y" | gh re --delete foo')).toMatchSnapshot()
        done()
    })
})
