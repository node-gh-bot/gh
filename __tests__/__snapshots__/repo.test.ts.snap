// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: Repo Module Test Create new repo \`gh re --new foo --init\` 1`] = `
"Creating a new repo on protoEvangelion/foo
https://github.com/protoEvangelion/foo
[hook] gh re --browser  --user protoEvangelion --repo foo
"
`;

exports[`E2E: Repo Module Test Delete repo \`gh re --delete foo\` 1`] = `
"Deleting repo protoEvangelion/foo
? Are you sure? This action CANNOT be undone. [y/N] [52D[52C[2K[G? Are you sure? This action CANNOT be undone. [y/N] y[53D[53C[2K[G? Are you sure? This action CANNOT be undone. [y/N] y[53D[53C
Successfully deleted repo.
"
`;

exports[`E2E: Repo Module Test Fork a repo \`gh re --fork prettier --user prettier\` 1`] = `
"Forking repo prettier/prettier into protoEvangelion/prettier
Successfully forked: https://github.com/protoEvangelion/prettier
"
`;

exports[`E2E: Repo Module Test List Repos \`gh re --list\` 1`] = `
"protoEvangelion/100-days-of-code
protoEvangelion/adopt-a-family
protoEvangelion/alloy-ui
"
`;

exports[`E2E: Repo Module Test List labels on a repo \`gh re --label --list --repo gh\` 1`] = `
"Listing labels on protoEvangelion/gh
--user
duplicate
enhancement
"
`;
